Firestore/core/src/firebase/firestore/core/view_snapshot.h

```diff
@@ -33,6 +41,31 @@ enum class DocumentViewChangeType {
   kMetadata
 };
 
+/** A change to a single document's state within a view. */
+class DocumentViewChange {
+ public:
+  DocumentViewChange() = default;
+
+  DocumentViewChange(FSTDocument* document, DocumentViewChangeType type)
```

Comment:
Note: on Android, the enum is contianed in `DocumentViewChange`, so the name is more like `DocumentViewChange::Type`. On Web client, the enum is global, but the name is shorter, just `ChangeType`.

Reply:
@wilhuff Any preference here?

Reply:
No strong preference.

On Android, making types global forces us to put them in separate files. For small types that are naturally bound to some other one, it makes sense to nest them, especially because they can be imported in the outer context. We could certainly follow that here.

`DocumentViewChangeType` seems like the worst of all possible options though. We should either nest to follow Android or make the global and follow Web's name.

Ideally nested or not the enum name would be the same.

Reply:
Renamed to match Android.